middleware/index.ts

```diff
@@ -37,77 +42,23 @@ function createLogger(requestID?: string): Logger {
     return logger;
 }
 
-function errorMessage(error: Error): any {
+function buildErrorResponse(err: Error): any {
+    // any error in user-function-space should be considered a 500 toerhwi
+    // ensure we send down application/json in this case
+    return buildResponse(err instanceof MiddlewareError ? err.code : INTERNAL_SERVER_ERROR_CODE, err.message, err.stack);
+}
+
+function buildResponse(code: string, response: any, extraInfo?: string): any {
     // any error in user-function-space should be considered a 500
     // ensure we send down application/json in this case
     return Message.builder()
         .addHeader('content-type', 'application/json')
-        .addHeader('x-http-status', '500')
-        .payload({
-            error: error.toString(),
-        })
+        .addHeader('x-http-status', code)
+        .addHeader('x-extra-info', extraInfo || '')
```

Comment:
Requested Fix: my preference would be to return a structured json object like the proxy currently does:

```
Content-Type: application/json

{"error": "Error text goes here", "extraInfo": "stack goes here"}
```

HTTP Headers have a restricted length (usu 2k) and don't deal well with characters like line breaks/tabs/ etc that could be found in a stack trace.

Reply:
Error is returned as `response`.